### DISCUSSION TITLE
eta_stripping / eta_plating

### DISCUSSION THREAD

**Discussion by onordlander** [pybamm-team/PyBaMM#2031] [Q&A] [answered]
Hello PyBaMM!

I'm implementing the irreversible lithium plating model and I seem to get Li-plating while the [Negative electrode lithium plating reaction overpotential [V]] is greater than 0, which I assume should only occur while I have negative values of the lithium plating reaction overpotential. 
Literature seems to refer to a li-plating reaction overpotential as: 

eta_plating = delta_phi - eta_sei

Source: https://pubs.rsc.org/en/content/articlelanding/2022/cp/d2cp00417h

In plating.py the lithium plating reaction overpotential seems to be defined as:

eta_stripping = delta_phi + phi_ref + eta_sei
eta_plating = -eta_stripping

What exactly is phi_ref in plating.py, and why is it included in the equation for lithium plating reaction overpotential?

**Comment by valentinsulzer** [comment]
@DrSOKane

**Comment by DrSOKane** [comment, marked as answer]
Thank you for your question @onordlander. 

The idea that plating only occurs when the electrolyte potential is greater than that of the electrode is a common misconception in the literature. Plating is accelerated under that condition, but also happens at a slower rate under more normal conditions.

If you go to examples/notebooks/models/lithium-plating.ipynb, it is demonstrated that there is always a small but finite amount of lithium plated, even at rest.

I am in the process of upgrading the lithium plating model with more features and hope to publish this to pybamm's develop branch within the next week.

Many thanks,
Dr. Simon O'Kane
Research Associate, Department of Mechanical Engineering, Imperial College London

**Reply by Yuxiao-Lan** [reply]
@DrSOKane Sorry to disturb you. I noticed that the overpotential for Li stripping in `plating.py`  is defined as the **sum** of the `electrode surface potential difference` and the` electrode SEI film overpotential` (`eta_stripping = delta_phi + eta_sei`). This definition appears to differ from the one provided in your paper (Okane 2022), where the overpotential for Li plating is expressed as $\eta_{Li}=\phi_{s}-\phi_{e}-\eta_{SEI}$, implying that it is the **difference** between the `electrode surface potential difference` and the `electrode SEI film overpotential`. Why the sign of `eta_sei` is different?

**Reply by DrSOKane** [reply]
Hi Yuxiao, the reason for the difference is that the `eta_SEI` used in PyBaMM has the opposite sign to the one used in O'Kane _et al._ (2022). The one in PyBaMM is defined on line 151 of base_kinetics.py:

`eta_sei = -j_tot * L_sei * R_sei`

The one in O'Kane _et al._ (2022) does not have the minus sign at the start, hence the difference.

**Reply by Yuxiao-Lan** [reply]
I got it! Thank you for the rapid response and explanation.

**Comment by onordlander** [comment]
Thank you for the rapid response and explanation @DrSOKane.

Could you please explain the added phi_ref term to the equation for the calculation of eta_stripping and eta_plating?

Thanks

**Reply by DrSOKane** [reply]
@onordlander phi_ref is a consequence of the way the electrolyte potential is de-dimensionalized. If you go to pybamm/models/submodels/electrolyte_conductivity/base_electrolyte_conductivity.py, you'll see where it comes from.

**Reply by onordlander** [reply]
Thank you once again @DrSOKane.

Looking forward to the new upcoming version of the lithium plating model.

Have a nice weekend!

**Reply by Mrzhang-hub** [reply]
@DrSOKane   Sorry to disturb you. I am confused that the code in pybamm is not correspond to the equations of the paper you have published before.  There are two exp functions in the paper, yet the code in pybamm only has one.  Besides, the prefactor in pybamm is not the same as the equation in your paper. Can you help me out?  
![paper equation](https://user-images.githubusercontent.com/59716473/166689866-5b20c29a-bb00-4471-a4f1-8be977e25819.png)
![pybamm code](https://user-images.githubusercontent.com/59716473/166689923-a5635247-4575-452d-944d-35780d0ca05d.png)

**Reply by DrSOKane** [reply]
@Mrzhang-hub The partially reversible lithium plating model is still not ready to publish yet. It was meant to be finished by now but there are still a few bugs to sort out. Sorry about this.

**Reply by Mrzhang-hub** [reply]
@DrSOKane   I am excited about the partially reversible lithium plating model. But what I wanna ask is the irreversible lithium plating model.  Is the irreversible lithium plating code in pybamm correspond to the paper you have published in 2020? It seems they are different. I have put the picures in above comment. Look forward to your explanation.

**Reply by DrSOKane** [reply]
The irreversible plating model is the same as the reversible plating model except that the first of the two terms is zero.

**Reply by Mrzhang-hub** [reply]
I got it! Thank u.

**Comment by amirsh213** [comment]
I see the results of the partially reversible Li plating simulation (and the reference paper) in Pybamm examples but I can't actually use it in the model. Does it mean it is not ready yet? 
If it's not ready yet, I was wondering if I simulate fast charge cycling for a long time with reversible plating, SEI growth, and the porosity change turned on with SEI, does it trigger irreversible plating at some point or it will be always reversible?
